ipa-core/src/report/hybrid.rs

```diff
@@ -757,6 +782,54 @@ where
     }
 }
 
+impl<BK, V> Shuffleable for IndistinguishableHybridReport<BK, V>
+where
+    BK: SharedValue + BooleanArray,
+    V: SharedValue + BooleanArray,
+{
+    // this requires BK:BAXX + B:BAYY  such that XX + YY <= 48
+    // this is checked in a debud_assert call in ::new below
+    // PERF OPPORTUNITY
+    // note that BA96 would likely be a better fit here. however, that would require a `BA128`
+    // in order to use `impl_malicious_shuffle_share!` and `BA128` cannot currently be
+    // implemented with `the boolean_array_impl!` macro as the trait `secret_sharing::Block`
+    // is not implemented for `bitvec::array::BitArray<[u8; 16]>`
```

Comment:
There is a macro `store_impl` to implement `Block`. The only tricky thing here is that `Block` is used both for boolean arrays and Galois field types, and they can't both attempt to implement `Block` for the same type.

Reply:
I think I have this working, but I'll double check I'm not duplicating it, and open it in a different PR. There's also a weird tradeoff here because we want it to be efficiently sized but also not too small for future implementations. we may end up wanting a macro for this implementation...